model/messages/consensus.go

```diff
@@ -1,16 +1,30 @@
 package messages
 
 import (
+	"github.com/onflow/flow-go/model"
 	"github.com/onflow/flow-go/model/flow"
 )
 
-// BlockProposal is part of the consensus protocol and represents the the leader
+// BlockProposal is part of the consensus protocol and represents the leader
 // of a consensus round pushing a new proposal to the network.
 type BlockProposal struct {
 	Header  *flow.Header
 	Payload *flow.Payload
 }
 
+// StructureValid checks basic structural validity of the message and ensures no required fields are nil.
+func (m *BlockProposal) StructureValid() error {
```

Comment:
+1

Reply:
I didn't add nil checks for top-level messages because the codec will never decode to a nil value (tested [here](https://github.com/onflow/flow-go/pull/3470/files#diff-e829a255cdabbac01eaed3c7e5b334703ad1e1e9972875073c44f3aa7e8b877aR16-R26)).